lib/Backend/Lower.cpp

```diff
@@ -17930,6 +17930,38 @@ Lowerer::GenerateArgOutForStackArgs(IR::Instr* callInstr, IR::Instr* stackArgsIn
     return saveLenInstr->GetDst()->AsRegOpnd();
 }
 
+void
+Lowerer::GenerateLoadStackArgumentByIndex(IR::Opnd *dst, IR::RegOpnd *indexOpnd, IR::Instr *instr, int32 offset, Func *func)
+{
+    // Load argument set dst = [ebp + index].
+
+    IR::RegOpnd *ebpOpnd = IR::Opnd::CreateFramePointerOpnd(func);
+    IR::IndirOpnd *argIndirOpnd = nullptr;
+
+    // The stack looks like this:
+    //       [new.target or FrameDisplay] <== EBP + formalParamOffset (4) + callInfo.Count - 1
+    //       arguments[n]                 <== EBP + formalParamOffset (4) + n
+    //       ...
+    //       arguments[1]                 <== EBP + formalParamOffset (4) + 2
+    //       arguments[0]                 <== EBP + formalParamOffset (4) + 1
+    //       this or new.target           <== EBP + formalParamOffset (4)
+    //       callinfo
+    //       function object
+    //       return addr
+    // EBP-> EBP chain
+
+    //actual arguments offset is LowererMD::GetFormalParamOffset() + 1 (this)
+
+    int32 actualOffset = GetFormalParamOffset() + offset;
+    Assert(GetFormalParamOffset() == 4);
```

Comment:
This is the one defined in the lowerer, not the in machine dependent Lower(er)MD classes. This one returns 4 always and is fine as the offset is guaranteed to be from ebp.